codegen/reflect_interface.go

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package codegen
+
+import (
+	"bytes"
+	"encoding/gob"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"text/template"
+
+	"github.com/golang/mock/mockgen/model"
+	"github.com/pkg/errors"
+)
+
+type reflectData struct {
+	ImportPath string
+	Symbols    []string
+}
+
+// ReflectInterface uses reflection to obtain interface information identified by symbols in given import path
+// projRoot is the root dir where mockgen is installed as a vendor package
+func ReflectInterface(projRoot string, importPath string, symbols []string) (*model.Package, error) {
```

Comment:
nit: `projRoot` and `importPath` are both `string`, you should use more concise `projRoot, importPath string, symbols []string`

nit: `symbols` is a `[]string` but only ever passed a single parameter. Should this be just a `string`?

Reply:
I tried to make it generic to extract multiple interfaces if needed, the current use cases happen to only need to extract one interface.